packages/block-editor/src/components/media-placeholder/index.js

```diff
@@ -130,8 +131,23 @@ export class MediaPlaceholder extends Component {
 				setMedia = onSelect;
 			}
 		} else {
+			const [ file ] = files;
+
+			if ( file.type === 'image/svg+xml' ) {
```

Comment:
Should we limit this logic to files smaller than a given amount of bytes?

Reply:
That sounds like a good idea. Like me check some browser limitations for data URIs.